### DISCUSSION TITLE
جلسه چهارم دوره بهترین مسیر ورود به دنیای برنامه‌نویسی

### DISCUSSION THREAD

**Discussion by alimd** [AliMD/alwatr#175] [Webinar & Meetups]
# دوره CIW و مبانی طراحی اپلیکیشن‌های تحت وب (بخش اول)

## جلسه چهارم دوره بهترین مسیر ورود به دنیای برنامه‌نویسی

`در تاریخ: ۸-۵-۱۴۰۱`

### سلام و عرض ادب، خیلی خوش‌آمدید

لطفا اگر مطلب مفیدی در رستای محتوای مطرح شده دارید همراه با تمرین‌های این جلسه در اینجا مطرح کنید.

[دریافت ویدیو وبینار](https://eseminar.tv/wb76239)

**مباحت مهم مطرح شده:**

۱. معرفی وب اپلیکیشن به عنوان هدف و آینده مباحث این دوره

۲. معرفی Progressive Web Application 
نمونه: https://github.com/hemanth/awesome-pwa#apps

۳. معرفی hybrid web applications
خیلی از اپلیکیشن‌های معروفی که روی گوشی نصب می‌کنید با تکنولوژی‌های وب ساخته شده‌اند مثلا
Twitter, Gmail, Apple App Store, Apple iBooks, Instagram, Evernote, Cryptochange, JustWatch, Untappd
برنامه‌های زیادی همچون vscode

۴. آموزش‌ها برای من سنگین هستش! چه کنم؟

۵. مفاهیم پایه:
- CIW
- Domain
- Host / Server / VPS
- DNS / DNS Providers
- Cloudflare

۶. ساخت اولین وبسایت شخصی در گیت‌هاب

**لینک‌های جانبی و مطالب تکمیلی:**

مقاله‌ای تکمیلی در مورد بخش‌های مختلف یک URL
[9 Parts of a URL You Should Know](https://medium.com/@joseph.pyram/9-parts-of-a-url-that-you-should-know-89fea8e11713)

مقالعه‌ای تکمیلی در مورد تفاوت URL / URI / URN
[The Difference Between a URL, URI, and a URN](https://danielmiessler.com/study/difference-between-uri-url/)

سایتی برای مقایسه DNS Provider های مختلف در دنیا
[dnsperf.com](https://www.dnsperf.com/)

ابزار آنلاین پینگ و لوکاپ و تریس و ...
[ping.eu](https://ping.eu/)

دستور dig آنلاین
[digwebinterface.com](https://www.digwebinterface.com/)

مقالات تکمیلی فارسی جهست درک بهتر مفاهیم سرور
[سرور چیست و چگونه کار می‌کند؟](https://www.arvancloud.com/blog/%d8%b3%d8%b1%d9%88%d8%b1-%da%86%db%8c%d8%b3%d8%aa/)
[VPS چیست؟](https://www.arvancloud.com/blog/vps-%da%86%db%8c%d8%b3%d8%aa/)
[سرور ابری چیست و چه مزایایی دارد؟](https://www.arvancloud.com/blog/%d8%b3%d8%b1%d9%88%d8%b1-%d8%a7%d8%a8%d8%b1%db%8c-%da%86%db%8c%d8%b3%d8%aa/)
[تفاوت هاست اشتراکی با سرور مجازی](https://www.arvancloud.com/blog/%d8%aa%d9%81%d8%a7%d9%88%d8%aa-%d9%87%d8%a7%d8%b3%d8%aa-%d8%a7%d8%b4%d8%aa%d8%b1%d8%a7%da%a9%db%8c-%d8%a8%d8%a7-%d8%b3%d8%b1%d9%88%d8%b1-%d9%85%d8%ac%d8%a7%d8%b2%db%8c/)

نگاهی به داخل دیتاسنترهای گوگل
[Inside a Google Datacenter](https://www.aparat.com/v/P7pj1)

آدرس github page من
[alimd.github.io](https://alimd.github.io/)

**[مطالب بیشتر در کانال تلگرام](https://t.me/AliMDSchool)**

**Comment by A-sina** [comment]
salam va dorood
[site](https://a-sina.github.io/)

**Comment by adltalab** [comment]
https://adltalab.github.io/

**Comment by alirezaFAkhar** [comment]
سلام استاد خسته نباشید 
https://alirezafakhar.github.io/

**Comment by sabermanesh** [comment]
[سایت گیت هاب](https://sabermanesh.github.io/)

**Comment by ghost** [comment]
https://MNKazem.github.io/

**Comment by amiresp** [comment]
https://amiresp.github.io/

**Comment by amir91179** [comment]
سلام استاد 
https://amir91179.github.io/

**Comment by berenjian** [comment]
https://berenjian.github.io

**Comment by mahshidhooshmand** [comment]
https://mahshidhooshmand.github.io/

**Comment by mahyasbm** [comment]
https://mahyasbm.github.io/

**Comment by amirmmolazemi** [comment]
# salam ostad mihandoost
### taklif jalase 4 va bazi jalase aval 
### [my site](https://amirmmolazemi.github.io/)
### [Game](https://output.jsbin.com/fiduxah)

**Comment by shosseinfarjadi** [comment]
سلام استاد تکالیف
https://shosseinfarjadi.github.io/

**Comment by marihooshmand** [comment]
https://marihooshmand.github.io/

**Comment by MKMoghadasan** [comment]
https://mkmoghadasan.github.io/

**Comment by AbolfazlRasoulizade** [comment]
https://abolfazlrasoulizade.github.io/

**Comment by Rahimpanah** [comment]
Hello teacher,
this is my exercise for session 4. I was busy, so I could not make it beautiful, I will do that as soon as possible.
Rahimpanah.github.io

Many thanks,
Reyhaneh

**Comment by M-Beheshti** [comment]
salam 
http://m-beheshti.github.io/

**Comment by KianooshSoleimani** [comment]
Hi there
https://kianooshsoleimani.github.io/

**Comment by sepehrislami** [comment]
https://sepehrislami.github.io

**Comment by MohammadHoseynSeyedi85** [comment]
https://mohammadhoseynseyedi85.github.io
+یک کمی تقلب از روی تکلیف قبلی

**Comment by SMsajjadSM** [comment]
my project site
https://github.com/SMsajjadSM/SMsajjadSM.github.io
and my site
SMsajjadSM.github.io

**Comment by armanasadian** [comment]
https://armanasadian.github.io/

**Comment by AmirrezaTN** [comment]
Salam
https://amirrezatn.github.io

**Comment by AlirezaKhodayari** [comment]
سلام  و عرض ادب
https://alirezakhodayari.github.io
